tests/test_slurm_launcher.py

```diff
@@ -0,0 +1,340 @@
+    launcher = SlurmLauncher(
+        launched={known_id: _LaunchedJobInfo("mock-id", "mock-job")}
+    )
+    unknown_id = create_job_id()
+
+    monkeypatch.setattr(launcher, "_stop_job", fail_if_called)
+    with pytest.raises(
+        errors.LauncherError, match=f"has not launched a job with id `{unknown_id}`"
+    ):
+        launcher.get_status(known_id, unknown_id)
+
+
+@pytest.mark.parametrize("is_het_job", [False, True])
+@pytest.mark.parametrize("scancel_rc", [0, 123])
+def test_slurm_launcher_stops_jobs(monkeypatch, is_het_job, scancel_rc):
+    if is_het_job:
+        monkeypatch.setenv("SLURM_HET_SIZE", "123456")
+    else:
+        monkeypatch.delenv("SLURM_HET_SIZE", raising=False)
+    monkeypatch.setattr(
+        "smartsim._core.launcher_.slurm.slurm_commands.scancel",
+        lambda *_, **__: (scancel_rc, "out", "err"),
+    )
+    id_ = create_job_id()
+    info = _LaunchedJobInfo("mock-id", "mock-job")
+    launcher = SlurmLauncher(launched={id_: info})
+    monkeypatch.setattr(
+        launcher,
+        "get_status",
+        lambda *ids: dict(zip(ids, itertools.repeat(JobStatus.CANCELLED))),
+    )
+    assert launcher.stop_jobs(id_) == {id_: JobStatus.CANCELLED}
+    assert info.status_override == (
+        JobStatus.CANCELLED if is_het_job and scancel_rc != 0 else None
+    )
+
+
+requires_slurm = pytest.mark.skipif(
+    shutil.which("srun") is None
+    or shutil.which("sbatch") is None
+    or shutil.which("sacct") is None
+    or shutil.which("scancel") is None,
+    reason="Slurm utilities could be found",
+)
+
+
+def requires_alloc_size(num_nodes):
+    try:
+        alloc_size = int(os.environ.get("SLURM_NNODES", None))
+    except (TypeError, ValueError):
+        alloc_size = None
+    return pytest.mark.skipif(
+        alloc_size is None or alloc_size < num_nodes,
+        reason=f"Test requires an allocation with at least {num_nodes} nodes",
+    )
+
+
+@requires_slurm
+@requires_alloc_size(1)
+def test_srun_hello_world(make_srun_command, test_dir):
+    launcher = SlurmLauncher()
+    srun = make_srun_command(
+        ["-N", "1", "-n", "3"], ["echo", "hello world"], use_current_alloc=True
+    )
+    id_ = launcher.start(srun)
+    time.sleep(1)
+    assert launcher.get_status(id_)[id_] == JobStatus.COMPLETED
+    with open(os.path.join(test_dir, "echo.out"), "r") as fd:
+        assert fd.read() == "hello world\n" * 3
+
+
+@pytest.mark.xfail(reason=r"Slurm launcher cannout parse `CANCELLED by \d+` syntax")
+@requires_slurm
+@requires_alloc_size(1)
+def test_srun_sleep_for_two_min_with_cancel(make_srun_command):
+    launcher = SlurmLauncher()
+    srun = make_srun_command(
+        ["-N", "1", "-n", "1"], ["sleep", "120"], use_current_alloc=True
+    )
+    id_ = launcher.start(srun)
+    time.sleep(1)
+    assert launcher.get_status(id_)[id_] == JobStatus.RUNNING
+    launcher.stop_jobs(id_)
+    time.sleep(1)
+    assert launcher.get_status(id_)[id_] == JobStatus.CANCELLED
```

Comment:
Question for reviewers: Is this something we want to be able to handle? SmartSim V0.8 also fails this test. It's impl will set the status to `JobStatus.FAILED` and this impl currently sets the status to `JobStatus.UNKOWN`.

If so, should this be handled in this ticket or a follow on?

Reply:
Do we care about the distinction between unknown and cancelled? If not, ignore it.  

If yes, new ticket. I wouldn't add more to your existing ticket since it's technically a breaking change and non-critical.

Reply:
Sounds like a plan! I think we care about the difference (or at least we cared enough to put it in the original API), so I'll add a note to my "tickets to write for slurm launcher" list!